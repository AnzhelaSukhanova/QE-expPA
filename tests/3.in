(set-option :produce-models true)
(set-logic BV)
(declare-fun y () (_ BitVec 4))
(assert
 (exists ((x (_ BitVec 4))) (and (and (or (bvult y (bvmul x #b0110)) (= y (bvmul x #b0110))) (or (bvult y (bvmul x #b0011)) (= y (bvmul x #b0011)))) (or (bvult y (bvmul x #b0010)) (= y (bvmul x #b0010))))))
