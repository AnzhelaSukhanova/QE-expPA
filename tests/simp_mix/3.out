(let (($e1 (bvadd y (bvadd (bvmul z #b0011) #b1000)))) (let (($e2 (bvadd (bvshl #b0001 y) (bvadd #b0100 (bvmul y #b1011))))) (let (($e3 (bvshl #b0011 y))) (let (($e4 (bvmul y #b1100))) (let (($e5 (= $e1 #b0000))) (let (($e6 (ite (bvult $e1 #b1000) (ite (bvult $e1 #b0100) (ite (bvult $e1 #b0010) (ite $e5 #b1111 #b0000) #b0001) #b0010) #b0011))) (let (($e7 (bvshl #b0001 $e6))) (let (($e8 (and (not (and (not (= $e4 $e6)) (not (bvult $e6 $e4)))) (not (and (not (= $e3 $e6)) (not (bvult $e3 $e6))))))) (let (($e9 (= $e3 #b0000))) (let (($e10 (and $e9 (= #b0001 $e1)))) (let (($e11 (and (not (and (not (= #b0001 $e4)) (not (bvult #b0001 $e4)))) (not (and (not (= #b0001 $e3)) (not $e9)))))) (let (($e12 (and $e11 (= $e1 #b0010)))) (let (($e13 (and (not (and (not (= $e4 #b0010)) (not (bvult #b0010 $e4)))) (not (and (not (= $e3 #b0010)) (not (bvult $e3 #b0010))))))) (let (($e14 (and $e13 (= $e1 #b0100)))) (let (($e15 (and (not (and (not (= #b0011 $e4)) (not (bvult #b0011 $e4)))) (not (and (not (= #b0011 $e3)) (not (bvult $e3 #b0011))))))) (let (($e16 (and $e15 (= #b1000 $e1)))) (let (($e17 (and (not (and (not (= #b0100 $e4)) (not (bvult #b0100 $e4)))) (not (and (not (= #b0100 $e3)) (not (bvult $e3 #b0100))))))) (let (($e18 (and $e17 $e5))) (let (($e19 (and (not (and (not (= $e4 #b0101)) (not (bvult #b0101 $e4)))) (not (and (not (= $e3 #b0101)) (not (bvult $e3 #b0101))))))) (let (($e20 (and $e19 $e5))) (let (($e21 (and (not (and (not (= $e4 #b0110)) (not (bvult #b0110 $e4)))) (not (and (not (= $e3 #b0110)) (not (bvult $e3 #b0110))))))) (let (($e22 (and $e21 $e5))) (let (($e23 (and (not (and (not (= #b0111 $e4)) (not (bvult #b0111 $e4)))) (not (and (not (= #b0111 $e3)) (not (bvult $e3 #b0111))))))) (let (($e24 (and $e23 $e5))) (let (($e25 (and (not (and (not (= #b1000 $e4)) (not (bvult #b1000 $e4)))) (not (and (not (= #b1000 $e3)) (not (bvult $e3 #b1000))))))) (let (($e26 (and $e25 $e5))) (let (($e27 (and (not (and (not (= $e4 #b1001)) (not (bvult #b1001 $e4)))) (not (and (not (= $e3 #b1001)) (not (bvult $e3 #b1001))))))) (let (($e28 (and $e27 $e5))) (let (($e29 (and (not (and (not (= $e4 #b1010)) (not (bvult #b1010 $e4)))) (not (and (not (= $e3 #b1010)) (not (bvult $e3 #b1010))))))) (let (($e30 (and $e29 $e5))) (let (($e31 (and (not (and (not (= #b1011 $e4)) (not (bvult #b1011 $e4)))) (not (and (not (= #b1011 $e3)) (not (bvult $e3 #b1011))))))) (let (($e32 (and $e31 $e5))) (let (($e33 (and (not (and (not (= #b1100 $e4)) (not (bvult #b1100 $e4)))) (not (and (not (= #b1100 $e3)) (not (bvult $e3 #b1100))))))) (let (($e34 (and $e33 $e5))) (let (($e35 (and (not (and (not (= $e4 #b1101)) (not (bvult #b1101 $e4)))) (not (and (not (= $e3 #b1101)) (not (bvult $e3 #b1101))))))) (let (($e36 (and $e35 $e5))) (let (($e37 (and (not (and (not (= #b1110 $e4)) (not (bvult #b1110 $e4)))) (not (and (not (= #b1110 $e3)) (not (bvult $e3 #b1110))))))) (let (($e38 (and $e37 $e5))) (let (($e39 (= $e4 #b1111))) (let (($e40 (and $e39 $e5))) (let (($e41 (and (bvult #b0001 $e1) $e9))) (let (($e42 (and (bvult #b0010 $e1) $e11))) (let (($e43 (and (bvult #b0100 $e1) $e13))) (let (($e44 (and (bvult #b1000 $e1) $e15))) (let (($e45 (and $e17 (not $e5)))) (let (($e46 (and $e19 (not $e5)))) (let (($e47 (and $e21 (not $e5)))) (let (($e48 (and $e23 (not $e5)))) (let (($e49 (and $e25 (not $e5)))) (let (($e50 (and $e27 (not $e5)))) (let (($e51 (and $e29 (not $e5)))) (let (($e52 (and $e31 (not $e5)))) (let (($e53 (and $e33 (not $e5)))) (let (($e54 (and $e35 (not $e5)))) (let (($e55 (and $e37 (not $e5)))) (let (($e56 (and $e39 (not $e5)))) (let (($e57 (= $e2 #b0000))) (let (($e58 (ite (bvult $e2 #b1000) (ite (bvult $e2 #b0100) (ite (bvult $e2 #b0010) (ite $e57 #b1111 #b0000) #b0001) #b0010) #b0011))) (let (($e59 (bvshl #b0001 $e58))) (let (($e60 (and (not (and (not (= $e4 $e58)) (not (bvult $e58 $e4)))) (not (and (not (= $e3 $e58)) (not (bvult $e3 $e58))))))) (let (($e61 (and (= #b0001 $e2) $e9))) (let (($e62 (and (= $e2 #b0010) $e11))) (let (($e63 (and (= #b0100 $e2) $e13))) (let (($e64 (and (= #b1000 $e2) $e15))) (let (($e65 (and $e57 $e17))) (let (($e66 (and $e57 $e19))) (let (($e67 (and $e57 $e21))) (let (($e68 (and $e57 $e23))) (let (($e69 (and $e57 $e25))) (let (($e70 (and $e57 $e27))) (let (($e71 (and $e57 $e29))) (let (($e72 (and $e57 $e31))) (let (($e73 (and $e57 $e33))) (let (($e74 (and $e57 $e35))) (let (($e75 (and $e57 $e37))) (let (($e76 (and $e57 $e39))) (let (($e77 (and (bvult #b0001 $e2) $e9))) (let (($e78 (and (bvult #b0010 $e2) $e11))) (let (($e79 (and (bvult #b0100 $e2) $e13))) (let (($e80 (and (bvult #b1000 $e2) $e15))) (let (($e81 (and (not $e57) $e17))) (let (($e82 (and (not $e57) $e19))) (let (($e83 (and (not $e57) $e21))) (let (($e84 (and (not $e57) $e23))) (let (($e85 (and (not $e57) $e25))) (let (($e86 (and (not $e57) $e27))) (let (($e87 (and (not $e57) $e29))) (let (($e88 (and (not $e57) $e31))) (let (($e89 (and (not $e57) $e33))) (let (($e90 (and (not $e57) $e35))) (let (($e91 (and (not $e57) $e37))) (let (($e92 (and (not $e57) $e39))) (and (not (and (not $e78) (not $e77) (not $e62) (not $e61) (not $e79) (not $e63) (not $e80) (not $e64) (not $e81) (not $e65) (not $e82) (not $e66) (not $e83) (not $e67) (not $e84) (not $e68) (not $e85) (not $e69) (not $e86) (not $e70) (not $e87) (not $e71) (not $e88) (not $e72) (not $e89) (not $e73) (not $e90) (not $e74) (not $e91) (not $e75) (not $e92) (not $e76) (not $e77) (not $e61) (not $e78) (not $e62) (not $e79) (not $e63) (not $e80) (not $e64) (not $e81) (not $e65) (not $e82) (not $e66) (not $e83) (not $e67) (not $e84) (not $e68) (not $e85) (not $e69) (not $e86) (not $e70) (not $e87) (not $e71) (not $e88) (not $e72) (not $e89) (not $e73) (not $e90) (not $e74) (not $e91) (not $e75) (not $e92) (not $e76) (not $e77) (not $e61) (not $e78) (not $e62) (not (and (bvult $e59 $e2) $e60)) (not (and $e60 (= $e2 $e59))))) (not (and (not $e42) (not $e41) (not $e12) (not $e10) (not $e43) (not $e14) (not $e44) (not $e16) (not $e45) (not $e18) (not $e46) (not $e20) (not $e47) (not $e22) (not $e48) (not $e24) (not $e49) (not $e26) (not $e50) (not $e28) (not $e51) (not $e30) (not $e52) (not $e32) (not $e53) (not $e34) (not $e54) (not $e36) (not $e55) (not $e38) (not $e56) (not $e40) (not $e41) (not $e10) (not $e42) (not $e12) (not $e43) (not $e14) (not $e44) (not $e16) (not $e45) (not $e18) (not $e46) (not $e20) (not $e47) (not $e22) (not $e48) (not $e24) (not $e49) (not $e26) (not $e50) (not $e28) (not $e51) (not $e30) (not $e52) (not $e32) (not $e53) (not $e34) (not $e54) (not $e36) (not $e55) (not $e38) (not $e56) (not $e40) (not $e41) (not $e10) (not $e42) (not $e12) (not (and (bvult $e7 $e1) $e8)) (not (and $e8 (= $e1 $e7))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))
